store/modules/generator/generator.helpers.ts

```diff
@@ -60,18 +60,25 @@ export const helpers = {
         let additionnalPath = '';
         // Images are not directly stored at the root level
         if (pathsNumber > 3) {
-          // Removing possible filename at the end of the URL or #
-          if (pathArray[pathArray.length - 1].indexOf('.') !== -1 || pathArray[pathArray.length - 1].indexOf('#') !== -1) {
+          // Removing possible filename at the end of the URL or # or duplication on the path if the icon src already has it
+          var index = 1;
+          while (pathArray[pathArray.length - index].indexOf('.') !== -1 || pathArray[pathArray.length - index].indexOf('#') !== -1 || icon.src.split('/')[1] === pathArray[pathArray.length - index]) {
```

Comment:
Could we use a for loop here? Im always a little afraid of a while loop without a `break` scenario, especially in this part of the codebase where we have to handle so many different edge cases

Reply:
Switched to for loop